### DISCUSSION TITLE
Trying to merge a tm and splitter after they are splitted

### DISCUSSION THREAD

**Discussion by citystrawman** [CGAL/cgal#8299] [Q&A] [answered]
## Issue Details

I am using split function to intersect two surface meshes(one tm and one splitter) , named slope and slide as follows:
![image](https://github.com/CGAL/cgal/assets/9004748/9e795794-bfa6-44e0-bf25-4316a0f4f0b4)
![image](https://github.com/CGAL/cgal/assets/9004748/8acec4df-3f19-4f77-b37b-5b245bbc7662)

and I want to combine the newly created mesh and the splitter into one mesh. To make the mesh no self intersection, my way is to do split tm and splitter interchangeably so that both tm and splitter are intersected(their intersected nodes' cordinates are exactly the same), and here are the screenshots after split: (the gree nodes are the new intersected nodes)
![image](https://github.com/CGAL/cgal/assets/9004748/9bd725d9-e993-4bf6-80c8-81715a7bcc97)
![image](https://github.com/CGAL/cgal/assets/9004748/bc3cce4d-bc27-4ce0-8e2b-aee1c9db1ec7)

and now, here comes my question: How can I merge the two meshes into one and make the intersected nodes of the two meshes that has same coordinates into one node?  what I can imagine is to compare the distance of the two nodes and merge the two that are extremely close. Or, is there a better and simpler way to do this? 

I appreciate if anyone could help me solve this issue.

**Comment by sloriot** [comment, marked as answer]
If I understand correctly, you can use `CGAL::Polygon_mesh_processing::corefine_and_compute_difference(slope, slide)` directly.

**Reply by citystrawman** [reply]
> If I understand correctly, you can use `CGAL::Polygon_mesh_processing::corefine_and_compute_difference(slope, slide)` directly.

Sorry, I tried `corefine_and_compute_difference(slope, slide)` like this [example](https://doc.cgal.org/latest/Polygon_mesh_processing/Polygon_mesh_processing_2corefinement_difference_remeshed_8cpp-example.html), however the output mesh is as follows:
![image](https://github.com/CGAL/cgal/assets/9004748/8030f304-b143-4c85-8c54-73f2f2662c97)

what I want is just the second image in my first post which is the combination of the splitted slope(3rd image) and splitted slide(4th image) where the intersected points (green points) being a single vertex.

**Reply by citystrawman** [reply]
> If I understand correctly, you can use `CGAL::Polygon_mesh_processing::corefine_and_compute_difference(slope, slide)` directly.

I just found that using `copy_face_graph(splited_slide, splited_slope) ` would result a union of the two:
![image](https://github.com/CGAL/cgal/assets/9004748/ad4e0f78-a741-4d34-b122-fb9d2fef7034)
However, is there a way to combine the intersected vertex that has the same coordinates into one?

**Comment by sloriot** [comment]
You need to reverse the orientation of slide, using `PMP::reverse_face_orientations()` before calling `corefine_and_compute_difference()`.

**Reply by citystrawman** [reply]
> You need to reverse the orientation of slide, using `PMP::reverse_face_orientations()` before calling `corefine_and_compute_difference()`.

I tried reverse slide orientation, however the result is not satisfactory because slide surface does not retain in this model:
![image](https://github.com/CGAL/cgal/assets/9004748/8b17c6a4-c0a6-427f-a41c-7473ecad2acb)
what I need is that slide retains in this model:
![image](https://github.com/CGAL/cgal/assets/9004748/6cd1609f-0f3c-4f00-b953-33dac60a0fce)

**Reply by citystrawman** [reply]
> You need to reverse the orientation of slide, using `PMP::reverse_face_orientations()` before calling `corefine_and_compute_difference()`.

BTW I just found stitch_borders method and it seems that the duplicated vertex are merged.:

![image](https://github.com/CGAL/cgal/assets/9004748/727ab0a3-6f05-4b1f-ac52-06400691c6ba)

![image](https://github.com/CGAL/cgal/assets/9004748/d61311b9-9aa7-4317-9b75-af1c07cf4cf1)

Although I am not sure whether all nodes are merged now (I will check it soon)

**Reply by sloriot** [reply]
What do you mean by it does not retain the slide surface? Are you expecting to keep all the slide, and not only the part below slope? 
In such a case you would end up with something non-manifold.

**Reply by citystrawman** [reply]
> What do you mean by it does not retain the slide surface? Are you expecting to keep all the slide, and not only the part below slope? In such a case you would end up with something non-manifold.

actually I just need only the part below slope if possible, but if there's no way to get off the part above slope, it is acceptable.

**Reply by citystrawman** [reply]
> What do you mean by it does not retain the slide surface? Are you expecting to keep all the slide, and not only the part below slope? In such a case you would end up with something non-manifold.

By the way what can I imply by your words "you would end up with something non-manifold"? will a non-manifold model result inconsistency in isotropic remesh ?(I just used isotropic remesh to the stitched model and the intersected edges between the two models turned inconsistency), or will it result something bad in generating tetrahedra?(which I will do it later)

**Reply by sloriot** [reply]
Surface_mesh cannot contain a non-manifold surface.
